utilities/post_migration.py

```
@@ -448,128 +445,122 @@ def check_static_ip_preservation(
+    ssh_conn = vm_ssh_connections.create(vm_name=vm_name, username=ssh_username, password=ssh_password)
+    last_missing: list[str] = []
 
-        LOGGER.info(f"Verifying interface {interface_name}: IP={expected_ip}, Subnet={expected_subnet}")
+    LOGGER.info(
+        f"Verifying {len(static_interfaces)} static IPs: "
+        f"{[(iface['name'], iface['ip_address']) for iface in static_interfaces]}"
+    )
 
-        # Get current network configuration via SSH with retry logic It takes time for secondary IPs to appear
-        ssh_conn = vm_ssh_connections.create(vm_name=vm_name, username=ssh_username, password=ssh_password)
+    def verify_all_static_ips() -> list[dict[str, Any]] | None:
```

Ruff (0.15.12)

[warning] Too many branches (14 > 12)

(PLR0912)